tests/test_loan.py

```diff
@@ -7,67 +7,103 @@
 # it under the terms of the MIT License; see LICENSE file for more details.
```

Comment:
Add more tests to test machines `if`, test when it should raise exceptions and test `CallableTransitions`